src/Raven.Server/Documents/ETL/Providers/AI/Embeddings/EmbeddingGenerationException.cs

```diff
@@ -0,0 +1,5 @@
+﻿using System;
+
+namespace Raven.Server.Documents.ETL.Providers.AI.Embeddings;
+
+public class EmbeddingGenerationException(string msg, Exception innerException) : Exception(msg, innerException);
```

Comment:
What value does this give us?

Reply:
We use dedicated exceptions in the load phase of each ETL. This way we can adjust exception handling and alerts specifically for a given ETL (not always we do it though). Anyway I think it's good practice to show a dedicated exception to a user, with our custom message explaining it often more detailed (we do it in Kafka). Inner exception is always the one from a dedicated API that we use.